cmd/nanotube/listen.go

```diff
@@ -86,6 +109,11 @@ func readFromConnection(wg *sync.WaitGroup, conn net.Conn, queue chan string, st
 			zap.String("sender", conn.RemoteAddr().String()))
 	}
 
+	scanForRecords(conn, queue, stop, ms)
+}
+
+func scanForRecords(r io.Reader, queue chan string, stop <-chan struct{}, ms *metrics.Prom) {
+	sc := bufio.NewScanner(r)
```

Comment:
UDP is not playing well with bufio.
The reason is that one read() from  UDP will return back one datagram.
If we use bufio, and buffer size(at bufio level)  is < length of datagram, the unread part is discarded&lost.
See also: https://linux.die.net/man/2/recv
In this case, I would not use any buffering from UDP, we could just call strings.split() by \n for data coming from UDP and call Read() by hand.

Reply:
I am not sure we are discussing the same problem but there is this thing
https://github.com/golang/go/blob/531b6d3126bb810dcdfcce14983cea32236c322d/src/bufio/scan.go#L188
the scanner buffer is sized to hold one token (one line in our case). Max is 64k.

Reply:
Do you think it will still cut the input buffer and we will lose the data?

Reply:
You should be able to reproduce this by connecting and sending one datagram > startBufSize.

Reply:
Yes, you are right. The scanner reads the startBufSize sized buffer. The buffer extension is only used if no token delimiter is encountered. It is not used to accommodate large one-time inputs.